test/stdlib/tgmath.swift

```diff
@@ -18,74 +18,31 @@ import StdlibUnittest
 
 let MathTests = TestSuite("TGMath")
 
-func print3(_ op: String, _ d: Double, _ f: Float, _ g: CGFloat) -> String {
-#if arch(i386) || arch(arm)
-  if (f != Float(g)  &&  f.isNaN != g.isNaN) {
-    return "\(op)(CGFloat) got \(g) instead of \(f)"
-  }
-#else
-  if (d != Double(g)  &&  d.isNaN != g.isNaN) {
-    return "\(op)(CGFloat) got \(g) instead of \(d)"
-  }
-#endif
-  return "\(op) \(d) \(f) \(op)"
+func expectEqualWithAccuracy(_ actualDouble: Double, _ expectedDouble: Double, ulps: Int, _ message: String = "", file: String = #file, line: UInt = #line) {
```

Comment:
I would probably make this a generic on FloatingPoint so we can re-use and update a single definition:
~~~~
internal func equalWithAccuracy<T>(_ observed: T, _ expected: T, ulps: T) -> Bool
  where T: FloatingPoint {
    if (observed == expected) || (observed.isNaN && expected.isNaN) {
      return true
    }
    return abs(observed - expected) <= expected.ulp * ulps
}
~~~~
or similar.

Reply:
I agree your `equalWithAccuracy<T>` would be useful in other tests, but `CGFloat` verification requires extra hoops.  To ensure that `CGFloat` versions match `Double` or `Float` you really need to have all three actuals available.

Reply:
I was thinking only of replacing expectEqualWithAccuracy with expectTrue(equalWithAccuracy<T>( ... )) for now.

Reply:
Of course.  Done.